Library/Homebrew/formula.rb

```diff
@@ -1013,9 +1018,10 @@ def _outdated_versions(options = {})
 
     installed_kegs.each do |keg|
       version = keg.version
+      tab = Tab.for_keg(keg)
       all_versions << version
 
-      return [] if pkg_version <= version && !version.head?
+      return [] if pkg_version <= version && epoch == tab.epoch && !version.head?
```

Comment:
`outdated_version` is actually `not_outdated_version`.

I don't really like `not` prefixes, but If you're OK with them I make the change. Also, `equal_epoch` can be done `same_epoch`.